presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java

```diff
@@ -114,35 +114,31 @@ public CassandraPartitionResult getPartitions(ConnectorTableHandle tableHandle,
             return ImmutableList.of();
         }
 
-        Set<List<Object>> partitionKeysSet = getPartitionKeysSet(table, tupleDomain);
+        List<Set<Object>> partitionKeysList = getPartitionKeysList(table, tupleDomain);
 
-        // empty filter means, all partitions
-        if (partitionKeysSet.isEmpty()) {
+        Set<List<Object>> filterList = Sets.cartesianProduct(partitionKeysList);
+        // empty filters means, all partitions
+        if (filterList.isEmpty()) {
             return cassandraSession.getPartitions(table, ImmutableList.of());
         }
 
-        ImmutableList.Builder<CassandraPartition> partitions = ImmutableList.builder();
-        for (List<Object> partitionKeys : partitionKeysSet) {
-            partitions.addAll(cassandraSession.getPartitions(table, partitionKeys));
-        }
-
-        return partitions.build();
+        return ImmutableList.copyOf(cassandraSession.getPartitions(table, partitionKeysList));
```

Comment:
why do you copy here and not in `122`?

Reply:
I was aiming to be inline with the original implementation. But since `getPartitions` already [returns](https://github.com/prestodb/presto/blob/144438c58a24967146c5da97a59d0bf6fc255e6c/presto-cassandra/src/main/java/com/facebook/presto/cassandra/NativeCassandraSession.java#L405) a list, I guess it's redundant.